File: solutions/1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    ht := make(map[int]int)
    
    for i, num := range nums {
        if j, exists := ht[target-num]; exists {
            return []int{j, i}
        }
        
        ht[num] = i
    }
    
    return []int{}
}

File: solutions/1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> ht = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int tmp = nums[i];

            if (ht.containsKey(target - tmp)) {
                return new int[] {ht.get(target - tmp), i};
            }

            ht.put(tmp, i);
        }
        
        return new int[0];
    }
}